### PR TITLE
feat(admin): add License details and Support debug dump

### PR DESCRIPTION
**Pull request by derrickmehaffy**
### What does it do?

Adds two admin features to make license management and support triage easier, both folded into the existing `Settings > Overview` page as their own cards rather than new nav entries.

**License details (Growth and Enterprise)**

- `Settings > Overview` gains a Plan card (Growth and Enterprise only) showing the status, license mode (online/offline), subscription ID, and the effective per-feature entitlements (workflows, stages per workflow, releases, content-history retention, etc.). Limits set to a very large sentinel value are shown as `Unlimited`, and day-based limits are humanized (for example `1500 days` renders as `~4 years`). Plan and seats are not repeated here since the Overview already shows them.
- Online licenses show a single `Last license validity check` line (re-validated every 12 hours) rather than the short-lived license validity date, because an online license does not carry the real subscription end date. Offline licenses show the real expiry from `license.txt`.
- A running trial badges as `In Trial` rather than `Active`, gets the community support links instead of the support portal, and is not shown the AI overage rate.
- `Unknown` distinguishes "couldn't reach the license registry, showing the last license we hold" from "the license was refused", and the seat count survives a license lapsing.
- Shows a warning when the instance is running on a cached license after a registry outage.
- Backend: exposes `strapi.ee.expireAt`, adds a `strapi.ee.entitlements` registry populated by the review-workflows, content-releases, content-history and audit-logs modules, and extends the `license-limit-information` endpoint with the new fields (with a `satisfies` contract check).

**Support debug dump (all editions, RBAC gated)**

- A Support card on `Settings > Overview`, with the snapshot action gated by a new `admin::debug-dump.read` permission, that generates a redacted diagnostic JSON export for the Support team: versions, system info, database, plugins, providers, content-model counts, project customizations, the full sanitized config, `package.json`, and an allowlisted subset of `process.env`.
- Secrets are masked by a layered scrubber (secret-named keys, secret-looking values, provider and DB credentials, app keys, salts). App and home directory paths are relativized, and the license key never leaves the server. The preview is read-only and can be copied or downloaded.
- A plan-aware callout links to the right help channels: Discord, GitHub Discussions and GitHub Issues for Community; Strapi Support and GitHub Issues for Growth and Enterprise.
- Adds a `strapi.getCustomizations()` detector that reports custom controllers, services and routes (beyond the default factories) and whether `src/index` was modified beyond the template.

_Screenshots:_

### Why is it needed?

Support has no fast way to see what a customer's license actually entitles them to, and collecting a customer's environment and config for triage takes a lot of back-and-forth. This surfaces the license entitlements directly in the admin panel and gives customers a one-click, pre-redacted export to share, which should cut triage time.

### How to test it?

1. Add an EE license to `examples/kitchensink-ts/license.txt`, then run `cd examples/kitchensink-ts && yarn develop`.
2. `Settings > Overview`: confirm the Plan card (Growth and Enterprise) shows the status, mode, subscription ID, seats, the entitlements, and (for an online license) the `Last license validity check` line.
3. `Settings > Overview`, Support card: click `Generate snapshot`, confirm the preview renders with secrets shown as `[REDACTED]`, and that `Copy` and `Download` work. Confirm the support callout matches the edition.
4. Download a dump and grep it for a known secret from `config/*` (for example `APP_KEYS`) to confirm it does not appear.
5. Confirm the diagnostic snapshot panel is hidden for an admin role without `admin::debug-dump.read`.
6. Community edition: confirm Overview shows no Plan card and the Support card shows the community links.

Automated coverage: `yarn test:unit`, `yarn test:front`, `yarn test:api tests/api/core/admin/admin-debug-dump.test.api.ts` (asserts real redaction on a live dump and the permission gate), `yarn test:ts`, `yarn lint`.

### CODE REVIEW

**Review by innerdvations — CHANGES_REQUESTED** [review]
The debug-dump scrubber looks good for `providerOptions` and the known `extraPaths`, but a few common credential shapes still leak outside those containers — e.g. `accessKeyId`, nested SMTP `auth.pass`, and a bare `key` holding something like `sk_live_….`

_AI-assisted comment._

**Comment by derrickmehaffy** [comment]
> The debug-dump scrubber looks good for `providerOptions` and the known `extraPaths`, but a few common credential shapes still leak outside those containers — e.g. `accessKeyId`, nested SMTP `auth.pass`, and a bare `key` holding something like `sk_live_….`
> 
> _AI-assisted comment._

Yeah I need to do a deeper scan of possible secrects

**Comment by derrickmehaffy** [comment]
Updated the design based on Anka's feedback in Slack: 
image

**Review by innerdvations — CHANGES_REQUESTED** [review]
Re-reviewed on current HEAD — prior scrub gaps look fixed. A few remaining items before merge:

1. **`projectType` still hardcodes Enterprise for all EE** — `strapi.projectType` in the dump is still `strapi.EE ? 'Enterprise' : 'Community'`, so Growth instances report as Enterprise. Support triage is exactly who this field is for — please derive it the same way as the admin (`getProjectType` + `planPriceId`). I left a failing unit case that sets `planPriceId` to a growth id and expects `Growth`.

2. **`redact.test.ts` TypeScript CI** — `scrub()` returns `unknown`, and the newer assertions access properties without a cast (`TS2571`). Please type those results the same way the earlier cases already do so `yarn test:ts` passes.

3. **Stale snapshots** — API permission and CLI list snapshots need updating for the new `admin::debug-dump` action/service/route — those CI failures look PR-caused, not flaky. Please refresh the affected snapshots in this PR.

Nice work closing the earlier scrub gaps (`accessKeyId`, `auth.pass`, provider token shapes). One residual (non-blocking): a bare `key` holding a non-provider-looking secret still passes through — probably fine to leave unless you have a concrete Support case; matching every `key` would over-redact.

_AI-assisted comment._

File: packages/core/admin/admin/src/pages/Settings/pages/ApplicationInfo/components/DiagnosticSnapshotModal.tsx [hunk 1]
```diff
@@ -0,0 +1,157 @@
...
+            {formatMessage({
+              id: 'Settings.debug-dump.snapshot-modal.title',
+              defaultMessage: 'Diagnostic snapshot',
+            })}
+          </Modal.Title>
+        </Modal.Header>
+        <Modal.Body>
+          {isReady ? (
+            <Flex direction="column" alignItems="stretch" gap={4}>
+              <Typography textColor="neutral600">
```
Comment (mariekirsch) [inline packages/core/admin/admin/src/pages/Settings/pages/ApplicationInfo/components/DiagnosticSnapshotModal.tsx]: Could we remove the textColor here so it defaults to the default font colour?

File: packages/core/admin/admin/src/pages/Settings/pages/ApplicationInfo/components/DiagnosticSnapshotModal.tsx [hunk 2]
```diff
@@ -0,0 +1,157 @@
...
+                {formatMessage({
+                  id: 'Settings.debug-dump.snapshot-modal.description',
+                  defaultMessage:
+                    "This snapshot describes how your project is built so a third party can reproduce the bug locally. It does not contain any of your app's content and no credentials.",
+                })}
+              </Typography>
+              <Typography variant="sigma" textColor="neutral600">
```
Comment (mariekirsch) [inline packages/core/admin/admin/src/pages/Settings/pages/ApplicationInfo/components/DiagnosticSnapshotModal.tsx]: same here - could we remove the textColor

File: packages/core/admin/admin/src/pages/Settings/pages/ApplicationInfo/components/DiagnosticSnapshotModal.tsx [hunk 3]
```diff
@@ -0,0 +1,157 @@
...
+                </Typography>
+              </Box>
+            </Flex>
+          ) : (
+            <Flex justifyContent="center" alignItems="center" padding={7}>
+              <Loader>Generating diagnostic snapshot</Loader>
+            </Flex>
+          )}
+        </Modal.Body>
+        <Modal.Footer>
```
Comment (mariekirsch) [inline packages/core/admin/admin/src/pages/Settings/pages/ApplicationInfo/components/DiagnosticSnapshotModal.tsx:138]: The `Copy` and `Download` currently sit on opposite edges of the modal, which is a by-product of not displaying the `Cancel` button. I think for continuity and simplicity, we can add the Cncel button, and wrap the copy/download buttons around a `<Flex gap={2}></Flex>`


Updated design:
Image

File: packages/core/admin/admin/src/pages/Settings/pages/ApplicationInfo/components/DiagnosticSnapshotModal.tsx [hunk 4]
```diff
@@ -0,0 +1,157 @@
...
+          <Button
+            variant="tertiary"
```
Comment (mariekirsch) [inline packages/core/admin/admin/src/pages/Settings/pages/ApplicationInfo/components/DiagnosticSnapshotModal.tsx]: Could we please make the variant `secondary`?

File: packages/core/admin/admin/src/pages/Settings/pages/ApplicationInfo/components/SupportCard.tsx [hunk 5]
```diff
@@ -0,0 +1,285 @@
...
+            <>
+              <Box background="neutral150" width="0.1rem" />
+              <Flex direction="column" alignItems="start" gap={3} flex="1 1 0%">
+                <Typography variant="sigma" textColor="neutral600">
+                  {formatMessage({
+                    id: 'Settings.application.support.diagnostic-snapshot.label',
+                    defaultMessage: 'diagnostic snapshot',
+                  })}
+                </Typography>
+                <Typography variant="pi" textColor="neutral600">
```
Comment (mariekirsch) [inline packages/core/admin/admin/src/pages/Settings/pages/ApplicationInfo/components/SupportCard.tsx]: Could we remove `variant="pi" textColor="neutral600"`?
I tested locally and it works better with the default font size/colour

File: packages/core/admin/admin/src/pages/Settings/pages/ApplicationInfo/components/SupportCard.tsx [hunk 6]
```diff
@@ -0,0 +1,285 @@
...
+      aria-label={formatMessage(
+        {
+          id: 'Settings.application.support.tile-label',
+          defaultMessage: 'Open {title} in a new tab',
+        },
+        { title: formatMessage(title) }
+      )}
+      display="block"
+      hasRadius
+      background="neutral100"
```
Comment (mariekirsch) [inline packages/core/admin/admin/src/pages/Settings/pages/ApplicationInfo/components/SupportCard.tsx]: Let's remove the background here

File: packages/core/admin/ee/admin/src/pages/SettingsPage/pages/ApplicationInfoPage/components/LicenseInfo.tsx [hunk 7]
```diff
@@ -0,0 +1,366 @@
...
+            <Flex direction="column" alignItems="start" gap={2} tag="dd">
+              <Status size="S" variant={statusBadge.variant}>
+                <Typography variant="omega" fontWeight="bold">
+                  {formatMessage(statusBadge.label)}
+                </Typography>
+              </Status>
+              {checkinLine && (
+                <Flex gap={1} alignItems="center">
+                  <ClockCounterClockwise width="1.2rem" height="1.2rem" fill="neutral500" />
+                  <Typography variant="pi" textColor="neutral600">
```
Comment (mariekirsch) [inline packages/core/admin/ee/admin/src/pages/SettingsPage/pages/ApplicationInfoPage/components/LicenseInfo.tsx]: Let's make this textColor="neutral500"

File: packages/core/admin/admin/src/pages/Settings/pages/ApplicationInfo/components/SupportCard.tsx [hunk 8]
```diff
@@ -0,0 +1,285 @@
...
+                      'Ask a question, report a bug, or contribute to the Strapi community.',
+                  }
+            )}
+          </Typography>
+        </Flex>
+        <Flex alignItems="stretch" gap={7}>
+          <Box flex="1 1 0%">
+            <Grid.Root gap={4}>
+              {tiles.map((tile) => (
+                <Grid.Item key={tile.id} col={6} xs={12} direction="column" alignItems="stretch">
```
Comment (mariekirsch) [inline packages/core/admin/admin/src/pages/Settings/pages/ApplicationInfo/components/SupportCard.tsx]: The tiles correctly switch between 1 and 2 columns based on the window width, but because they’re sharing the card with the diagnostic section, the breakpoint is too low and it breaks beforehand.
Suggest sizing the tile grid off the tiles panel itself instead of the viewport

Image

**Comment by mariekirsch** [comment]
https://github.com/strapi/strapi/blob/4a4101264d7098754df36e85fa629fd2f2349d8c/packages/core/admin/ee/admin/src/pages/SettingsPage/pages/ApplicationInfoPage/components/AdminSeatInfo.tsx#L88-L102

These links can be removed since they have been replaced by the _Manage_/_View subscription_ button
image

File: packages/core/admin/ee/admin/src/pages/SettingsPage/pages/ApplicationInfoPage/components/AIUsage.tsx
Comment (mariekirsch) [inline packages/core/admin/ee/admin/src/pages/SettingsPage/pages/ApplicationInfoPage/components/AIUsage.tsx]: Could we change this block to:

```
<Flex direction="row" alignItems="baseline" gap={1}>
    <Typography>{`${usedCredits.toFixed(2)}`}</Typography>
    <Typography variant="pi" textColor="neutral600">{`/ ${totalCredits}`}</Typography>
</Flex>
<Flex width="100%">
     <StyledProgressBar value={percentRemaining} size="M" />
</Flex>
```

File: packages/core/admin/ee/admin/src/pages/SettingsPage/pages/ApplicationInfoPage/components/AIUsage.tsx [hunk 9]
```diff
@@ -75,7 +69,13 @@ export const AIUsage = () => {
             </Typography>
```
Comment (mariekirsch) [inline packages/core/admin/ee/admin/src/pages/SettingsPage/pages/ApplicationInfoPage/components/AIUsage.tsx]: Could we change this block to:

```
<Flex direction="row" alignItems="baseline" gap={1}>
  <Typography variant="epsilon" textColor="danger600">
      {`${overage.toFixed(2)}`}
   </Typography>
    <Typography variant="pi" textColor="neutral600">
       {`/ ${totalCredits}`}
      </Typography>
</Flex>
<Flex width="100%">
   <StyledProgressBar value={percentOverage} size="M" color="danger" />
</Flex>
```

File: packages/core/admin/ee/admin/src/pages/SettingsPage/pages/ApplicationInfoPage/components/AIUsage.tsx [hunk 10]
```diff
@@ -75,7 +69,13 @@ export const AIUsage = () => {
...
           </>
         )}
+        <Typography variant="pi" textColor="neutral600">
+          {formatMessage({
+            id: 'Settings.application.ai-usage.overage-rate',
+            defaultMessage: '+$1.50 per 100 credits thereafter',
```
Comment (mariekirsch) [inline packages/core/admin/ee/admin/src/pages/SettingsPage/pages/ApplicationInfoPage/components/AIUsage.tsx]: I'm so sorry Derrick, I realised that I hadn't accounted for the overages state. Could we make a slight adjustment to the copy to:

> +$1.50 per 100 credits above plan limit

File: packages/core/admin/admin/src/translations/en.json [hunk 11]
```diff
@@ -113,6 +113,9 @@
   "Settings.adminTokens.editPage.title": "Edit Admin Token",
   "Settings.adminTokens.emptyStateLayout": "You don't have any content yet...",
   "Settings.adminTokens.title": "Admin Tokens",
+  "Settings.application.admin-seats": "Admin seats",
+  "Settings.application.ai-usage": "AI Usage",
+  "Settings.application.ai-usage.overage-rate": "+$1.50 per 100 credits thereafter",
```
Comment (mariekirsch) [inline packages/core/admin/admin/src/translations/en.json]: Could we make a slight adjustment to the copy to:

> +$1.50 per 100 credits above plan limit

File: packages/core/admin/admin/src/pages/Settings/pages/ApplicationInfo/components/DiagnosticSnapshotModal.tsx (hunk 1 above)
Comment (derrickmehaffy) [reply packages/core/admin/admin/src/pages/Settings/pages/ApplicationInfo/components/DiagnosticSnapshotModal.tsx]: Done in 8d4fe2d892.

File: packages/core/admin/admin/src/pages/Settings/pages/ApplicationInfo/components/DiagnosticSnapshotModal.tsx (hunk 2 above)
Comment (derrickmehaffy) [reply packages/core/admin/admin/src/pages/Settings/pages/ApplicationInfo/components/DiagnosticSnapshotModal.tsx]: Done in 8d4fe2d892.

File: packages/core/admin/admin/src/pages/Settings/pages/ApplicationInfo/components/DiagnosticSnapshotModal.tsx (hunk 4 above)
Comment (derrickmehaffy) [reply packages/core/admin/admin/src/pages/Settings/pages/ApplicationInfo/components/DiagnosticSnapshotModal.tsx]: Done in 8d4fe2d892.

File: packages/core/admin/admin/src/pages/Settings/pages/ApplicationInfo/components/SupportCard.tsx (hunk 6 above)
Comment (derrickmehaffy) [reply packages/core/admin/admin/src/pages/Settings/pages/ApplicationInfo/components/SupportCard.tsx]: Done in 19c8d88a8b.

File: packages/core/admin/admin/src/pages/Settings/pages/ApplicationInfo/components/SupportCard.tsx (hunk 5 above)
Comment (derrickmehaffy) [reply packages/core/admin/admin/src/pages/Settings/pages/ApplicationInfo/components/SupportCard.tsx]: Done in 19c8d88a8b.

File: packages/core/admin/ee/admin/src/pages/SettingsPage/pages/ApplicationInfoPage/components/LicenseInfo.tsx (hunk 7 above)
Comment (derrickmehaffy) [reply packages/core/admin/ee/admin/src/pages/SettingsPage/pages/ApplicationInfoPage/components/LicenseInfo.tsx]: Done in 2ce350dce4.

File: packages/core/admin/ee/admin/src/pages/SettingsPage/pages/ApplicationInfoPage/components/AIUsage.tsx
Comment (derrickmehaffy) [reply packages/core/admin/ee/admin/src/pages/SettingsPage/pages/ApplicationInfoPage/components/AIUsage.tsx]: Done in fbe7000f0c.

File: packages/core/admin/ee/admin/src/pages/SettingsPage/pages/ApplicationInfoPage/components/AIUsage.tsx (hunk 10 above)
Comment (derrickmehaffy) [reply packages/core/admin/ee/admin/src/pages/SettingsPage/pages/ApplicationInfoPage/components/AIUsage.tsx]: Done in fbe7000f0c.

File: packages/core/admin/admin/src/translations/en.json (hunk 11 above)
Comment (derrickmehaffy) [reply packages/core/admin/admin/src/translations/en.json]: Done in fbe7000f0c, both the component default and en.json.

File: packages/core/admin/admin/src/pages/Settings/pages/ApplicationInfo/components/DiagnosticSnapshotModal.tsx (hunk 3 above)
Comment (derrickmehaffy) [reply packages/core/admin/admin/src/pages/Settings/pages/ApplicationInfo/components/DiagnosticSnapshotModal.tsx:138]: Added Cancel and wrapped Copy/Download in a `Flex gap={2}` in 8d4fe2d892. Cancel is a `Modal.Close` so it matches how LogoInput does it. Matches your mockup.

File: packages/core/admin/admin/src/pages/Settings/pages/ApplicationInfo/components/SupportCard.tsx (hunk 8 above)
Comment (derrickmehaffy) [reply packages/core/admin/admin/src/pages/Settings/pages/ApplicationInfo/components/SupportCard.tsx]: Good catch, that was sizing off the viewport so it broke while the window was still wide.

Fixed in 19c8d88a8b, but not with a container query. `@container` is the direct way to say this and I tried it first, but jsdom cannot parse it and it fails the admin suite's stylesheet parse. Content-manager's `FormLayout` has a TODO about hitting the same wall.

So it is a wrapping flex row instead: a 50% flex-basis caps a row at two tiles no matter how wide the card gets, and `min-width: 16rem` is what forces the collapse to one column. Both are measured against the tile panel, so the switch is driven by the space the tiles actually have. Same behaviour you asked for, no test-env cost. Worth a look when you re-test since the collapse point is now 16rem per tile rather than a named breakpoint.

File: packages/core/admin/ee/admin/src/pages/SettingsPage/pages/ApplicationInfoPage/components/AIUsage.tsx (hunk 9 above)
Comment (derrickmehaffy) [reply packages/core/admin/ee/admin/src/pages/SettingsPage/pages/ApplicationInfoPage/components/AIUsage.tsx]: Done in fbe7000f0c, with one change to your snippet worth flagging.

`color="danger"` on that bar was doing nothing. `StyledProgressBar` hardcodes its own fill colours, and `ProgressBar` forwards unknown props to the underlying Radix div, so `color` was landing as an HTML `color` attribute and never reaching the bar. The overage bar has been rendering neutral grey this whole time, not red. I switched it to a transient `$variant` prop so it actually goes red.

Also moved the sentence copy we dropped ("X credits used above the Y credits available in your plan") onto both bars as an `aria-label`. "12.00 / 100" on its own does not tell a screen reader anything.

**Comment by derrickmehaffy** [comment]
> These links can be removed since they have been replaced by the _Manage_/_View subscription_ button

Removed in 2ce350dce4, along with the `admin-seats.support` and `admin-seats.add-seats` translation keys since nothing else used them. Updated the two tests that asserted those links.

**Comment by derrickmehaffy** [comment]
Thanks Marie, all 13 review comments are addressed across 8d4fe2d892, 19c8d88a8b, 2ce350dce4, fbe7000f0c and 4e381ae55d. `test:front` is green in both CE and EE, plus `test:ts:front`, lint and prettier.

Two things I did differently from the literal comment, both explained inline: the tile grid uses flex rather than a container query because jsdom cannot parse `@container`, and the overage progress bar needed an actual fix to turn red since `color="danger"` was inert.

**On your QA doc, admin seats on Enterprise: that was a real bug, fixed in 4e381ae55d.** The seat count was gated on the plan resolving to Growth, so any Enterprise license carrying a seat limit never showed one. Null `permittedSeats` just means an old license the registry has not re-issued, which the component already handles by rendering nothing, so the plan gate was redundant on top of being wrong. Worth re-testing with a license that has seats set.

**Comment by derrickmehaffy** [comment]
Follow-up from QA, pushed in `cdabe7ad7d` and `a1edfb1a62`.

- Trials get the community links instead of the Strapi support portal, and no longer see the AI overage rate. Neither applies to a trial.
- A running trial now badges as `In Trial` rather than `Active`.
- `Unknown` was a catch-all. It now says which of the two situations it is: we couldn't reach the license registry and are showing the last license we hold, or the license was refused. Only the second needs the customer to do anything. The cached-license warning this page was supposed to show turned out not to be rendering at all, so that's fixed too.
- Admin seats now show on any license that carries a seat limit, rather than only on Growth.

One thing I looked at and deliberately left alone: `isTrial` flipping to false once a license stops validating. That flag is what the trial-ended banner and modal key off, so making it sticky would suppress both.

The remaining QA items need a change on the licensing service side and are tracked separately.

**Review by innerdvations — CHANGES_REQUESTED** [review]
Re-reviewed at `be6665ee6480`. The three things from my July 30 pass are all fixed — dump `projectType` goes through `getProjectType`, `redact.test.ts` type-checks, and the permission/CLI snapshots are refreshed. The Overview fold reads well and the scrubber coverage is solid.

Two things about the expired-license path before this goes in.

**1. The debug dump reports an expired license as Community**

`strapi.EE` is `ee.enabled`, which `disable()` flips to false while wiping `licenseInfo` down to the key. `debug-dump.ts` gates `edition`, `projectType` and the entire `license` block on it, so an instance whose license lapsed produces a dump that is indistinguishable from a project that never had a license — no plan, no subscription id, no expiry, no entitlements.

That is the exact case Support is most likely to receive a dump for, and it contradicts the Overview page in this same PR, which correctly shows "Enterprise / Expired" off `retainedLicense`. The typed surface already has what the dump needs (`strapi.ee.licenseStatus` and `strapi.ee.retainedLicense`), so this can fall back the same way `licenseLimitInformation` does. Including `licenseStatus` in the payload would also let Support tell active from expired without inferring it from dates.

Failing proof, against this head:

```js
// packages/core/admin/server/src/services/__tests__/debug-dump.test.ts
it('includes retained license details when the license is expired (EE disabled)', async () => {
  const strapi = makeStrapi();
  strapi.EE = false;
  strapi.ee.planPriceId = undefined;
  strapi.ee.expireAt = undefined;
  strapi.ee.subscriptionId = undefined;
  strapi.ee.seats = undefined;
  strapi.ee.type = undefined;
  strapi.ee.licenseStatus = 'expired';
  strapi.ee.retainedLicense = {
    type: 'gold',
    isTrial: false,
    seats: 10,
    subscriptionId: 'sub_1',
    expireAt: '2026-01-01T00:00:00.000Z',
    planPriceId: 'cms-growth-monthly',
    features: [{ name: 'sso' }],
  };

  const dump = await debugDumpService({ strapi }).generate();

  expect(dump.strapi.projectType).toBe('Growth'); // Received: "Community"
  expect(dump.license).toBeDefined();
  expect(dump.license?.subscriptionId).toBe('sub_1');
});
```

**2. Retained `seats` has no consumer**

`licenseLimitInformation` returns `seats: strapi.ee.seats ?? retained?.seats ?? null`, and the comment above it says the fallback is there "so an expired/unknown license can still show its plan, seats and subscription". But `AdminSeatInfoEE` reads `permittedSeats`, which is the live enforcement value and is null once EE is disabled — so the seat row disappears on exactly the licenses the fallback was added for, and nothing else in the admin reads `seats`.

Either wire the display path to `seats` (keeping `permittedSeats` for the OVER_LIMIT tooltip and `shouldStopCreate`), or drop `seats` from the response and the comment. Happy either way — I just don't think it should ship claiming one and doing the other.

```tsx
// packages/core/admin/ee/admin/src/pages/SettingsPage/pages/ApplicationInfoPage/components/tests/AdminSeatInfo.test.tsx
test('Shows retained seats when live permittedSeats is wiped after expiry', () => {
  useLicenseLimits.mockReturnValue({
    ...LICENSE_MOCK,
    license: {
      ...LICENSE_MOCK.license,
      permittedSeats: null,
      seats: 10,
      licenseStatus: 'expired',
      enforcementUserCount: 4,
    },
  });

  const { getByText } = render(<AdminSeatInfoEE />);

  expect(getByText('Admin seats')).toBeInTheDocument(); // not found
  expect(withMarkup(getByText)('4/10')).toBeInTheDocument();
});
```

**Non-blocking:** the description still describes `Debug & support` as its own settings page and the paired Last/Next check-in, both of which the Overview fold replaced. Also, `superpowers/` in the root `.gitignore` looks unrelated to this work.

_AI-assisted comment._

**Comment by derrickmehaffy** [comment]
Both fixed, plus the two non-blocking notes.

**1. Expired licence reported as Community in the dump** — `dc445cd3e4`. You were right, and it did contradict the Overview page in this same PR. The dump now reads through to `retainedLicense` the way `licenseLimitInformation` does, and carries `licenseStatus` so active and expired are distinguishable without inferring from dates. Your failing case is in as a test, with extra assertions on `type`, `seats` and `features` coming off the retained snapshot.

One deliberate difference: `edition` still reports `strapi.EE ? 'EE' : 'CE'`. An expired instance genuinely is running as CE, and `edition: CE` alongside `projectType: Growth` and `licenseStatus: expired` is the honest description of that state. Happy to change it if you'd rather it tracked the licence instead.

I also had to give the test's `makeStrapi` a `licenseStatus`, and the CE case now sets `licenseStatus: 'none'` — that is what a real CE instance has, and without it the mock was relying on `undefined` to mean "no licence".

**2. Retained `seats` had no consumer** — `776f48343f`. Took the wire-it-up option: the display falls back to `seats` when `permittedSeats` is wiped, so the row survives a lapse. Enforcement is untouched, `licenseLimitStatus` and `shouldStopCreate` still run off `permittedSeats` server-side. Your test is in, plus one asserting the row stays hidden when neither value is present, so the pair actually discriminates.

**Non-blocking:** description updated — the Debug & support page and the paired Last/Next check-in were both replaced by the Overview fold, and I added the trial and `Unknown` changes from the QA pass. `superpowers/` removed from `.gitignore` in `832fc732f3`; that was local tooling noise that shouldn't have been on this branch.

Also in that last commit: `renewalDate` is typed as epoch ms rather than a string. It is currently always null because the registry has never populated it; that is being fixed separately, and epoch ms is the format it will arrive in, matching `expireAt` in the same payload.

CE and EE suites both at 821 passing, `test:ts` front and back clean.

**Review by innerdvations — CHANGES_REQUESTED** [review]
I just pushed some unit tests to show you the issue, hope that's ok — `ad14ee2` adds two failing cases. Heads up that this turns the backend unit job red until the two bugs below are fixed; it isn't a flake.

The two blockers from my last pass are fixed — expired instances now report their retained plan and license block in the dump, and the Plan card keeps the seat count after a lapse. Thanks for those. Two things left, both on the same "license lapsed" path the retention work targets.

**1. `licenseMode` still reads only the live license type**

`licenseLimitInformation` derives offline mode from the live type:

```ts
const licenseMode: 'online' | 'offline' =
  strapi.ee.type === 'gold' && process.env.STRAPI_DISABLE_LICENSE_PING?.toLowerCase() === 'true'
    ? 'offline'
    : 'online';
```

`disable()` wipes `licenseInfo.type` and keeps it on `retainedLicense`, so an expired offline gold instance reports `licenseMode: 'online'`. The Plan card then shows the "Last license validity check" line instead of "License valid until {expireAt}" — the real expiry from `license.txt`, which is the one thing an offline customer needs at that moment. Every neighbouring field already handles this, including `type` itself further down (`strapi.ee.type ?? retained?.type ?? null`), so it's just the mode check that misses it.

Suggested: resolve the type once (`const licenseType = strapi.ee.type ?? retained?.type;`) and use it for both `licenseMode` and `data.type`.

**2. The debug-dump path relativizer is POSIX-only**

```ts
if (appRoot && (value === appRoot || value.startsWith(`${appRoot}/`))) {
  return `<app>${value.slice(appRoot.length)}`;
}
if (homeDir && (value === homeDir || value.startsWith(`${homeDir}/`))) {
```

Both prefix checks require a forward slash, so on Windows nothing matches: `os.homedir()` and the app root come back as `C:\Users\<name>\...` and absolute paths ship verbatim in the snapshot. That's the app/home path relativization the PR advertises, silently not applying on a supported platform — and dumps go to Support, so it's a real (if mild) disclosure of usernames and directory layout.

Suggested: normalize separators before the prefix comparison, or accept `path.sep` as well as `/`.

The pushed tests are:

- `packages/core/admin/ee/server/src/controllers/__tests__/admin.license.test.ts` — expired gold + `STRAPI_DISABLE_LICENSE_PING=true` expects `offline`, gets `online`
- `packages/core/admin/server/src/utils/debug-dump/__tests__/redact.test.ts` — Windows app/home paths expect `<app>\src\index.js`, get the untouched absolute path

Rework or drop them if you'd rather assert this differently — they're just the shortest way I could show the two paths.

Everything else held up on re-read: the RBAC gate on `GET /admin/debug-dump` with its 403 integration test, the license key never entering the payload, the env allowlist, the wholesale masking of `server.app.keys` and the DB connection, the `ee.enabled` gate on `entitlements.list()`, and the unauthenticated `/admin/project-type` response still omitting license state.

_AI-assisted comment._

**Review by innerdvations — DISMISSED** [review]
Code LGTM now! Still needs QA, and make sure Product has signed off on this before merging

**Review by innerdvations — APPROVED** [review]
